sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobClient.java

```diff
@@ -91,38 +95,85 @@ public String getBlobServiceUrl() {
+    }
 
-        return storageBlobServiceClient.createContainerWithRestResponse(containerName,
-            timeout,
-            metadata,
-            publicAccessType,
-            cancellationToken);
+    /**
+     * Marks the specified container for deletion. The container and any blobs contained within it are later deleted
+     * during garbage collection. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/delete-container">Azure Docs</a>.
+     *
+     * @param containerName The container name.
+     */
+    @NonNull
+    public Void deleteContainer(@NonNull String containerName) {
+        return this.deleteContainerWithResponse(new ContainerDeleteOptions(containerName)).getValue();
+    }
+
+    /**
+     * Marks the specified container for deletion. The container and any blobs contained within it are later deleted
+     * during garbage collection. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/delete-container">Azure Docs</a>.
+     *
+     * @param options {@link ContainerDeleteOptions}
+     * @return The response information returned from the server when deleting a container.
+     */
+    @NonNull
+    public ContainerDeleteResponse deleteContainerWithResponse(@NonNull ContainerDeleteOptions options) {
+        Objects.requireNonNull(options);
+        return storageBlobServiceClient.deleteContainerWithRestResponse(options.getContainerName(),
+            options.getTimeout(), options.getRequestConditions(), options.getCancellationToken());
+    }
+
+    /**
+     * Returns the container's metadata and system properties. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/get-container-metadata">Azure Docs</a>.
+     *
+     * @param containerName The container name.
+     * @return The container's properties.
+     */
+    /* TODO: (gapra) This should probably return a handwrapped type? */
+    @NonNull
+    public ContainerGetPropertiesHeaders getContainerProperties(@NonNull String containerName) {
+        return this.getContainerPropertiesWithResponse(new ContainerGetPropertiesOptions(containerName)).getDeserializedHeaders();
+    }
+
+    /**
+     * Returns the container's metadata and system properties. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/get-container-metadata">Azure Docs</a>.
+     *
+     * @param options {@link ContainerGetPropertiesOptions}
+     * @return The response information returned from the server when getting a container's properties.
+     */
+    @NonNull
+    public ContainerGetPropertiesResponse getContainerPropertiesWithResponse(@NonNull ContainerGetPropertiesOptions options) {
+        Objects.requireNonNull(options);
+        BlobRequestConditions requestConditions = options.getRequestConditions() == null ? new BlobRequestConditions()
```

Comment:
Should we check that only lease id is set here?

Reply:
Related: I think the param validation should move from the impl class to this one

Reply:
I don't think it matters that much if the required parameter validation happens here or in Impl, but the latter would mean we don't have to duplicate code I guess. Is there a particular reason you think that should be here? I think that if we are thinking the Impl class is going to be auto-generated then it makes sense to have validations outside of it.

Reply:
Question: is `leaseId` the only required parameter in the options bag? Is `containerName` a part of it as well?

Reply:
lease Id is the only other service related optional param for this API.

Reply:
Yeah, I think since it will eventually be generated, we should do param validation on the client

Reply:
If `leaseId` is required then I would add it to the options bag constructor and validate that its not null there instead of on the method in the client.

Reply:
leaseId is optional. Sorry if I mis-communicated before.

Reply:
No problem. Even if `leaseId` is the only other value the service expects, since its optional I don't think we should validate its contents.

Reply:
We're not validating it's contents, just making sure the RequestConditions object exists before we call requestConditions.getLeaseId